### PR TITLE
Latency reduction in run fusion and rework pointwise schedule.

### PR DESCRIPTION
**Pull request by csarofeen**

### CODE REVIEW

File: torch/csrc/jit/codegen/cuda/expr_evaluator.h [hunk 1]
```diff
@@ -43,15 +43,15 @@ class TORCH_CUDA_API EvaluationContext {
 // Evaluates expressions in a Fusion IR, using the passed in
 // context (EvaluationContext) to query for concrete_values. The
 // evaluation context may override concrete values in the IR as well.
-class TORCH_CUDA_API ExpressionEvaluator : private IterVisitor {
+class TORCH_CUDA_API ExpressionEvaluator : protected IterVisitor {
```
Comment (jjsjann123) [inline torch/csrc/jit/codegen/cuda/expr_evaluator.h:46]: Noticed that we changed everything from `private` to `protected` but we don't have a new class that inherits from `ExpressionEvaluator`.

File: torch/csrc/jit/codegen/cuda/scheduler.cpp [hunk 2]
```diff
@@ -185,15 +187,19 @@ bool scheduleFusion(Fusion* fusion, const at::ArrayRef<c10::IValue> inputs) {
...
+      //
+      // if (!disable_unroll && tv->nDims() == 3) {
+      //   tv->axis(-2)->parallelize(ParallelType::Unroll);
+      //   tv->axis(-1)->parallelize(ParallelType::TIDx);
+      // } else {
+      //   if (tv->nDims() == 2)
+      //     tv->axis(-1)->parallelize(ParallelType::TIDx);
+      // }
     }
     TensorView* out0 = fusion->outputs()[0]->as<TensorView>();
```
Comment (jjsjann123) [inline torch/csrc/jit/codegen/cuda/scheduler.cpp:204]: This is not your problem, but the code below (`out0` & computing `numel`) is obsolete and we can just delete them all as well as the commented out unrolling code.

File: torch/csrc/jit/codegen/cuda/scheduler.cpp [hunk 3]
```diff
@@ -15,7 +15,7 @@ namespace jit {
 namespace fuser {
 namespace cuda {
 
-constexpr int kUnrollFactor = 4;
+constexpr int kUnrollFactor = 1;
```
Comment (jjsjann123) [inline torch/csrc/jit/codegen/cuda/scheduler.cpp:18]: Same here, if we are removing unrolling code, we can do it cleaner and getting rid of the macro.

**Review by jjsjann123 — APPROVED** [review]
LGTM

File: torch/csrc/jit/codegen/cuda/scheduler.cpp (hunk 3 above)
Comment (csarofeen) [reply torch/csrc/jit/codegen/cuda/scheduler.cpp:18]: We will want to re-enable this sooner than later.

**Comment by csarofeen** [comment]
broke this up into smaller PRs

**Pull request closed without merging**